### PR TITLE
Fix issue #7262 

### PR DESCRIPTION
**Pull request by JFCote**
The problem was that camelCase naming was forced only in this part of the code when everywhere else it is configurable.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09)

### Description of the PR

Fix issue #7262

This bug seems to have been introduced by this commit: https://github.com/swagger-api/swagger-codegen/commit/a63e3f14f58cdce57be0f1091905d8653eb26626
So I'm tagging its author for review: @defmonk0 

In summary, I've made sure that the parameters names in the path received the same treatment everywhere else instead of applying a camelCase by default.

### CODE REVIEW

**Comment by defmonk0** [comment]
The issue I see with this change is it essentially reverts to the original functionality before a63e3f1 (it would simply use the loop instead of the original regex method). This would simply cause the issue I was attempting to fix.

As described in [my response](https://github.com/swagger-api/swagger-codegen/issues/7262#issuecomment-355565524) to issue #7262, it seems we need to change this code to be compatible with the `modelPropertyNaming` option. The functionality of this option is described in #6632. 

My change is compatible when the option is not set (defaults to `camelCase`, I believe, per Angular style guide). This change is compatible when set to `original`, as described in #7262. We are completely missing compatibility with the option set to `PascalCase`.

Pinging @wing328, since he merged my changes initially and is active on every mention of this option I can find. ;P I imagine he'll have a better notion of how we're supposed to be supporting this, since I didn't even know it existed before now. Lol.

**Comment by JFCote** [comment]
@defmonk0 My change is compatible with the naming option since it uses `toVarName` which is used in all other generators to take care of this:
See here: https://github.com/swagger-api/swagger-codegen/pull/7313/files#diff-ad60e100e050d5ef52943803a2adc5fdR279

By calling this, it will check what is the prefered naming convention and replace the param name by the good naming in the path.

**Comment by defmonk0** [comment]
@JFCote Holy crap, I completely missed that function call nested inside the `.append`. My bad. Lol.

With that there, yes, I agree, this should work perfectly well. Looks fine to me.

**Comment by wing328** [comment]
@JFCote thanks for the quick fix.

@defmonk0 thanks for reviewing the change.
